lib/elixir/lib/access.ex

```diff
@@ -31,83 +26,52 @@ defmodule Access do
       iex> star_ratings[1.5]
       "★☆"
 
-  Note that the dynamic lookup syntax (`term[key]`) roughly translates to
-  `Access.get(term, key, nil)`.
+  The bracket syntax itself places no restrictions on the key types, but the data
+  structure being accessed may have such restrictions (ie. maps can have any key type,
+  but keyword lists must have atoms as keys).
 
-  `Access` can be combined with `Kernel.put_in/3` to put a value
-  in a given key:
+  This works well for dynamic data structures where keys can come and go over the
+  lifetime of a program _but is disabled by default for structs_, which have keys
+  defined at _compile-time_ (see: [Structs and Dot Syntax](#module-structs-and-dot-syntax)).
 
-      iex> map = %{a: 1, b: 2}
-      iex> put_in map[:a], 3
-      %{a: 3, b: 2}
+  ### Bracket Syntax and Access
 
-  This syntax is very convenient as it can be nested arbitrarily:
+  When compiling  source code, Elixir transforms each invocation of bracket syntax
+  into individual calls to `Access.get/2`.
 
-      iex> users = %{"john" => %{age: 27}, "meg" => %{age: 23}}
-      iex> put_in users["john"][:age], 28
-      %{"john" => %{age: 28}, "meg" => %{age: 23}}
+      iex> quote do a[:b] end
+      {{:., [], [Access, :get]}, [], [{:a, [], AccessTest}, :b]}
 
-  Furthermore, `Access` transparently ignores `nil` values:
+  Because `Access.get/2` is translated into `Access.get/3` with `nil` as the default,
+  accessing a non-existing key will _not raise_, but will instead return a `nil` value.
 
       iex> keywords = [a: 1, b: 2]
       iex> keywords[:c][:unknown]
       nil
 
-  Since `Access` is a behaviour, it can be implemented for key-value
-  data structures. The implementation should be added to the
-  module that defines the struct being accessed. `Access` requires the
-  key comparison to be implemented using the `===` operator.
-
-  ## Static lookups
-
-  The `Access` syntax (`data[key]`) cannot be used to access fields in
-  structs, since structs do not implement the `Access` behaviour by
-  default. It is also a design decision: the dynamic access lookup
-  is meant to be used for dynamic key-value structures, like maps
-  and keywords, and not by static ones like structs (where fields are
-  known and not dynamic).
-
-  Therefore Elixir provides a static lookup for struct fields and for atom
-  fields in maps. Imagine a struct named `User` with a `:name` field.
-  The following would raise:
+  Structs do not by default implement Access behaviour and so by default cannot use this
```

Comment:
I believe this is repeated information. We have just gone over this point at the end of the previous section?